backend/FwLite/LcmCrdt.Tests/Changes/UseChangesTests.cs

```diff
@@ -45,27 +46,54 @@ public async Task CanAddAllChangeTypes()
             var serializedQueuedChanges = JsonSerializer.Serialize(queuedChanges, Options);
             throw new Exception($"Failed to add changes: {e.Message}. JSON:\n\n{serializedQueuedChanges}", e);
         }
+    }
+
+    [Fact]
+    public async Task CanSyncAllChangesWithDuplicates()
+    {
+        var shuffledChangesWithDependencies = random.Shuffle(GetAllChanges())
+            .ToDictionary(change => change.Change, change => change.Dependencies == null ? null : random.Shuffle(change.Dependencies).ToList());
+        var pendingChanges = shuffledChangesWithDependencies.Select(c => c.Key).ToList();
+        var committedChanges = new List<IChange>(pendingChanges.Count);
+
+        while (pendingChanges is not [])
+        {
+            var change = FindFirstSatisfiedChangeRecursive(pendingChanges, shuffledChangesWithDependencies, committedChanges);
+            if (!pendingChanges.Remove(change)) throw new InvalidOperationException("Change not found in pending changes");
 
+            await fixture.DataModel.AddChange(Guid.NewGuid(), change);
+
+            // Add a duplicate change
+            // but we obviously can't create entities with the same ID
+            if (IsCreateChange(change)) change.EntityId = Guid.NewGuid();
```

Comment:
I'm not sure what might happen because we're modifying a change we previously gave to harmony. I suggest we clone the change (convert to json and back would be fine) to avoid any potential issues now or in the future.